src/board.rs

```diff
@@ -116,16 +117,18 @@ impl fmt::Display for Board {
         for rank in (0..self.squares.len() / self.width).rev() {
             write!(f, "\t")?;
             for file in 0..self.width {
-                let square = Square {
+                let square = UncheckedSquare {
```

Comment:
We don't need to validate here, since we know it's bounded already. We can just keep using squares.